scout/server/blueprints/alignviewers/controllers.py

```diff
@@ -14,6 +14,22 @@
 CUSTOM_TRACK_NAMES = ["Genes", "ClinVar", "ClinVar CNVs"]
 
 
+def set_session_tracks(display_obj):
+    """Save igv tracks as a session object. This way it's easy to verify that a user is requesting one of these files from remote_static view endpoint
+
+    Args:
+        display_obj(dict): A display object containing case name, list of genes, lucus and tracks
+    """
+    session_tracks = list(display_obj.get("reference_track", {}).values())
+    for key, track_items in display_obj.items():
+        if key not in ["tracks", "custom_tracks", "sample_tracks"]:
+            continue
+        for track_item in track_items:
+            session_tracks += list(track_item.values())
```

Comment:
Hm, one would need a plan to purge old session tracks as well, otherwise we will rather soon have users unable to log in due to overfull session stores.

Reply:
I don't think overfull session stores would happen anyway, since that session object doesn't contain much info, try to printout the content! And the session["igv_tracks"] gets overwritten every time you get to the alignments of a new case.

The object is a list of strings with the tracks for that case, not a dict with track data for each case you visit

Reply:
I've been investigating a solution and found this:: https://stackoverflow.com/questions/48994440/execute-a-function-after-flask-returns-response

I've pushed some code to this PR. Basically after the request is returned from igv or sashimi_igv, the session["ivg_tracks"] key/value gets removed